batch.ts

```diff
@@ -0,0 +1,114 @@
+import {transact, getMutationID, setMutationID} from './db';
+import {gcal} from './gcal';
+import {userError, ErrorKind} from './errors';
+
+export async function batch(payload: any, auth: string) {
+    validatePayload(payload);
+
+    let infos = [];
+    for (let m of payload.mutations) {
+        await transact(async (db) => {
+            const currentMutationID = await getMutationID(db, payload.clientID);
+            const expectedMutationID = currentMutationID + 1;
+
+            if (m.id > expectedMutationID) {
+                throw userError(`mutation.id is too high - expected ${expectedMutationID}`);
```

Comment:
I believe there is some inconsistency in batch endpoint implementations. Some places we say that the next ID must be greater then the current one and some places we say that I must be exactly one more. I think allowing it to be greater instead of exactly one more is safer in the sense of less likely to deadlock the protocol. however it also means that if the data layer goes back in time then we mistakenly play new mutations On Top of Old State in the data layer. I'm inclined to say that we should allow the mutation IDs to not be strictly sequential just increasing